### ISSUE TITLE
Make including files more convenient

### ISSUE THREAD

**Issue by Xandaros** [INPStarfall/Starfall#213]
It is very inconvenient, to use --@include probably/very/long/path/to/script.txt and then require("probably/very/long/path/to/script.txt").
Every time you have to make the decision of either grabbing your mouse, and copy that string or type it out all again. If you have just a few includes, this gets very annoying.
#2 attempts to fix that, but we have since found too many issues with that approach. The most apparent is probably: What if the original owner of the chip has since left the server?

Another idea is to add another preprocessor directive that includes the file and automatically prepends the script with a require statement.
If people want to load it manually or use dofile, they can use the existing directive.
Maybe also add the ability to upload whole directories, if you separate out your files for the sake of overview(e.g. if you use classic OOP principles and give each class its own file, not terribly uncommon, is it?), you can easily hit a wall of inconvenience when having to include all those files.

This approach is probably a little questionable, but at the moment including files is not very pleasant.
Especially when including lots of files, or using nested includes

**Comment by NotAkitake** [comment]
I agree with that, I find it kind of annoying to have to include and then require, it's a two step thing whereas in most languages, you simply include it and it does the rest of the work for you.
I guess in lua that would not be exactly possible (or not similar) but we can get close by doing it the way you propose.

"Another idea is to add another preprocessor directive that includes the file and automatically prepends the script with a require statement.
If people want to load it manually or use dofile, they can use the existing directive."

I think that would be just fine :)
